### PR TITLE
doc: add overview and getting started instructions

### PR DESCRIPTION
**Pull request by fabxc**

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -1,30 +1,28 @@
...
-└────────────┴────┬────┘                   └────────────┴────┬────┘
-                  │                                          │
-                Blocks                                     Blocks
-                  │                                          │
-                  v                                          v
-              ┌──────────────────────────────────────────────────┐
-              │               Google Cloud Storage               │
-              └──────────────────────────────────────────────────┘
-```
+* Data sources
```
Comment (Bplotka) [inline docs/design.md]: s/Data/Metric/? (: Maybe worth to be more specific here?

File: docs/design.md [hunk 2]
```diff
@@ -1,30 +1,28 @@
...
+* Stores
+* Queriers
```
Comment (Bplotka) [inline docs/design.md]: What about other things? Compactor, ruler?

File: docs/design.md [hunk 3]
```diff
@@ -1,30 +1,28 @@
...
 
-At the foundation of PromLTS stands the vanilla Prometheus server itself. Integration into the overall systems happens through a sidecar. No changes to the general setup should be necessary.
+Thanos provides two components that act as data sources: the Prometheus sidecar and rule nodes.
 
-Prometheus writes "blocks" of data, which represent all time series data collected during a fixed time window. A block is a simple directory with a few files:
+The sidecar implements the gRPC service on top of Prometheus' [HTTP and remote-read APIs][prom-http-api]. The rule node directly implements it on top of the Prometheus storage engine it is running.
+
+#### Data Backup
+
+Data sources that persist their data for long-term storage do so via the Prometheus 2.0 storage engine. The storage engine periodically produces immutable blocks of data for a fixed time range. A block is a directory with a handful of larger files containing all sample data and peristed indices that are required to retrieve the data:
```
Comment (Bplotka) [inline docs/design.md]: do they persist via Prometheus 2.0 storage engine or actually bucket syncing mechanism (shipper)?

File: docs/design.md [hunk 4]
```diff
@@ -33,143 +31,130 @@ Prometheus writes "blocks" of data, which represent all time series data collect
...
-└── tombstones
+└── meta.json
 ```
 
+
 A blocks top-level directory is a ULID (like UUID but lexicographically sortable and encoding the creation time).
 
 * Chunk files hold a few hundred MB worth of chunks each. Chunks for the same series are sequentially aligned. Series in return are aligned by their metric name. This becomes relevant further down.
 * The index file holds all information needed to lookup specific series by their labels and the positions of their chunks.
 * `meta.json` holds meta information about a block like stats, time range, and compaction level.
```
Comment (Bplotka) [inline docs/design.md]: Worth to add that it includes and addition to std meta.json -> external labels.

File: docs/design.md [hunk 5]
```diff
@@ -33,143 +31,130 @@ Prometheus writes "blocks" of data, which represent all time series data collect
...
+The store node is aware of the files' layout and translates data requests into a plan of a minimum amount of object storage request. Each requests may fetch up to hundreds of thousands of chunks at once. This is essential to satisfy even big queries with a limited amount of requests to the object storage.
 
-As block storage is generally slow, it is not feasible to always fetch data from it ad-hoc. Especially index searches, which have to sequentially read from multiple random positions in the index file to satisfy a query, would be very slow. Thus, an intermediate "store" layer is added, which then implements the same API as the sidecars.
+Currently only index data is cached. Chunk data could be cached but is orders of magnitude larger in size. In the current state, fetching chunk data from the object storage already only accounts for a small fraction of end-to-end latency. Thus, there's currently no incentive to increase the store nodes resource requirements/limit its scalability by adding chunk caching.
 
+### Stores & Data Sources - It's all the same
```
Comment (Bplotka) [inline docs/design.md]: :tada:

File: docs/design.md [hunk 6]
```diff
@@ -33,143 +31,130 @@ Prometheus writes "blocks" of data, which represent all time series data collect
...
-  * transfer: free within the zone
-  * operations: most ops will be GET ($0.004 per 10k), still unknown but probably minor, assume $200/month for now (500 million ops)
-* HA store node pair
-  * 2 x 64TB provisioned SSD: $21,760
-  * OR 2 x 64TB provisioned HDD: $5120
-  * 2 x n1-standard-32: $1,552
-* Query nodes:
-  * 4 x n1-standard-16: $1,552
+Thereby, storing 100 TB of metric data would cost approximately $2400/month on top of the baseline Prometheus setup.
+In return, being able to reduce the retention time of Prometheus instances from weeks to hours will provide cost savings for local SSD or network block storage (typically $0.17/GB).
```
Comment (Bplotka) [inline docs/design.md]: plus, memory cost reduction

File: docs/design.md [hunk 7]
```diff
@@ -33,143 +31,130 @@ Prometheus writes "blocks" of data, which represent all time series data collect
...
-  * storage: $15,000/month
-  * transfer: free within the zone
-  * operations: most ops will be GET ($0.004 per 10k), still unknown but probably minor, assume $200/month for now (500 million ops)
-* HA store node pair
-  * 2 x 64TB provisioned SSD: $21,760
-  * OR 2 x 64TB provisioned HDD: $5120
-  * 2 x n1-standard-32: $1,552
-* Query nodes:
-  * 4 x n1-standard-16: $1,552
+Thereby, storing 100 TB of metric data would cost approximately $2400/month on top of the baseline Prometheus setup.
```
Comment (Bplotka) [inline docs/design.md]: Worth to say approximately what does it mean to have 100TB of metrics (with tsdb compression).. We have almost 6 weeks within 170GB (with Prom HA), so we will potentially reach 100TB... with 69 years? xD $2400 looks scarier than it is if someone does not feel how much 100TB really is...

File: docs/overview.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+## Getting started
+
+### Requirements
+
+* One or more [Prometheus](https://prometheus.io) installations
+* An object storage bucket (optional)
+
+### Get Thanos!
+
+Thanos has no official relases yet. With a working installation of the Go toolchain, Thanos can be downloaded and built by running
```
Comment (Bplotka) [inline docs/overview.md]: `s/relases/releases`

File: docs/overview.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+The query layer is a stateless and horizontally scalable component of Thanos. It can find sources of metric data, such as the deployed sidecars, and evaluate PromQL expressions against their data.
+
+It hosts the Prometheus querying UI and implements its HTTP API, which allows tools such as Grafana to use it just like a regular Prometheus server.
+
+* [How to deploy query nodes](components/query#deployment)
+
+After a query node is deployed and connected to the cluster, it provides a global view of all available data. This means metrics from different Prometheus servers can be combined within a single query.
+
+It can deduplicate data that was collected from high-availablity pairs of Prometheus servers and thereby seamlessly closes gaps caused by downtime of single instances.
```
Comment (Bplotka) [inline docs/overview.md]: `s/high-availablity/high-availability`

File: docs/overview.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+Rule nodes can evaluate known Prometheus rules against the Thanos' querying layer. They can be deployed in a highly-available manner and metrics they produce are instantly visible within Thanos.
+
+* [How to deploy rule nodes](components/rule#deployment)
+
+### Compactor
+
+Metric data stored in the object storage is in the same format as on Prometheus' local disk. Similarly to compaction processes done by Prometheus, Thanos' can greatly benefit from data in the object storage being compacted.
+
+The compactor is a singleton components that continously ensures that data blocks in the object storage are consolidated, which reduces data size and increases query efficiency.
```
Comment (Bplotka) [inline docs/overview.md]: `s/components/component`

File: docs/design.md (hunk 2 above)
Comment (fabxc) [reply docs/design.md]: Ruler is a data source. Compactor is not a clustered component. 
I tried to keep the design doc a bit more abstract and don't pin it down to specific implementation we provide too much as that's already covered sufficiently in component docs.

Comment (Bplotka) [reply docs/design.md]: ah, right

File: docs/design.md (hunk 3 above)
Comment (fabxc) [reply docs/design.md]: Persistence happens through Prometheus. The shipper really just does dumb uploading – now with some extra logic to not reupload etc.

File: docs/design.md (hunk 7 above)
Comment (fabxc) [reply docs/design.md]: I already dialed it down from 750TB before :P But yea, will add some figures.

**Comment by fabxc** [comment]
Addressed comments

**Pull request merged**